src/Agent.Plugins/TestFilePublisher/Plugin/TestFilePublisherLogPlugin.cs

```diff
@@ -0,0 +1,212 @@
+        {
+            try
+            {
+                _logger = _logger ?? new TraceLogger(context);
+                _telemetry = _telemetry ?? new TelemetryDataCollector(new ClientFactory(context.VssConnection), _logger);
+
+                await PopulatePipelineConfig(context);
+
+                if (DisablePlugin(context))
+                {
+                    _telemetry.AddOrUpdate(TelemetryConstants.PluginDisabled, true);
+                    await _telemetry.PublishCumulativeTelemetryAsync();
+                    return false; // disable the plugin
+                }
+
+                _testFilePublisher = _testFilePublisher ??
+                                     new TestFilePublisher(context.VssConnection, PipelineConfig, new TestFileTraceListener(context), _logger, _telemetry);
+                await _testFilePublisher.InitializeAsync();
+                _telemetry.AddOrUpdate(TelemetryConstants.PluginInitialized, true);
+            }
+            catch (Exception ex)
+            {
+                context.Trace(ex.ToString());
+                _logger?.Warning($"Unable to initialize {FriendlyName}.");
+                if (_telemetry != null)
+                {
+                    _telemetry.AddOrUpdate(TelemetryConstants.InitializeFailed, ex);
+                    await _telemetry.PublishCumulativeTelemetryAsync();
+                }
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <inheritdoc />
+        public async Task ProcessLineAsync(IAgentLogPluginContext context, Pipelines.TaskStepDefinitionReference step, string line)
+        {
+            await Task.CompletedTask;
+        }
+
+        /// <inheritdoc />
+        public async Task FinalizeAsync(IAgentLogPluginContext context)
+        {
+            using (var timer = new SimpleTimer("Finalize", _logger, TimeSpan.FromMinutes(2),
+                new TelemetryDataWrapper(_telemetry, TelemetryConstants.FinalizeAsync)))
+            {
+                try
+                {
+                    await _testFilePublisher.PublishAsync();
+                }
+                catch (Exception ex)
+                {
+                    _logger.Info($"Error: {ex}");
+                    _telemetry.AddOrUpdate("FailedToPublishTestRuns", ex);
+                }
+            }
+
+            await _telemetry.PublishCumulativeTelemetryAsync();
+        }
+
+        /// <summary>
+        /// Return true if plugin needs to be disabled
+        /// </summary>
+        private bool DisablePlugin(IAgentLogPluginContext context)
+        {
+            // do we want to log that the plugin is disabled due to x reason here?
+            if (context.Variables.TryGetValue("Agent.ForceEnable.TestFilePublisherLogPlugin", out var forceEnable)
+                && string.Equals("true", forceEnable.Value, StringComparison.OrdinalIgnoreCase))
+            {
+                return false;
```

Comment:
>return false; [](start = 16, length = 13)

is there a reason to log the reason to disable...

Reply:
we can log with debug verbosity, itll only show up if system.debug=true

---
In reply to: [263715683](https://github.com/Microsoft/azure-pipelines-agent/pull/2141#discussion_r263715683) [](ancestors = 263715683)